Onboarding/Connector/Scripts/Enable-ImpactReporting.ps1

```diff
@@ -0,0 +1,287 @@
+      Test-FeatureRegistration $Namespace $FeatureName
+      Log "Successfully registered feature - $FeatureName in namespace - $Namespace"
+  }
+  else {
+      Log "Feature $FeatureName in namespace $Namespace is already registered"
+  }
+
+  Log "Getting the changes for the feature $FeatureName in namespace $Namespace propagated to the subscription"
+  Register-Namespace $Namespace
+  Log "Changes for the feature $FeatureName in namespace $Namespace are propagated to the subscription"
+}
+
+
+function Test-FeatureRegistration {
+  param (
+      [string]$Namespace,
+      [string]$FeatureName
+  )
+
+  Log "Verifying registration for feature: $FeatureName in namespace $Namespace, feature registration can take upto several minutes"
+  $RegistrationStatus = ""
+  $AttemptCount = 1
+  $SleepDuration = 10
+  while ($RegistrationStatus -ne "Registered") {
+      Log "Attempt #${AttemptCount}: Verifying registration status for $FeatureName in namespace $Namespace"
+      $RegistrationStatus = (Get-AzProviderFeature -ProviderNamespace "$Namespace" -FeatureName "$FeatureName").RegistrationState
+      $SleepDuration = $SleepDuration * 2
+      Log "Attempt #${AttemptCount}: Waiting for $SleepDuration seconds before retrying."
+      Start-Sleep -Seconds $SleepDuration
+      $AttemptCount = $AttemptCount + 1
+  }
+}
+
+function Add-PermissionsForAlertReading {
+  param (
+      [string]$SubscriptionId
+  )
+
+  Log "Setting up permissions for alert reading"
+  $RoleName = "Azure-Alerts-Reader-Role" # DND
+  $ConnectorAppName = "AzureImpactReportingConnector"
+
+  $RoleId = (Get-AzRoleDefinition -Name $RoleName).id
+  if (-not $RoleId) {
+      Log "Creating custom role for alert reading."
+      $CustomRoleJsonFilePath = "/tmp/impact-reporting-role.json"
+      CreateCustomRoleJson $SubscriptionId $RoleName | Out-File $CustomRoleJsonFilePath
+      $RoleId = (New-AzRoleDefinition -InputFile $CustomRoleJsonFilePath).id
+      Log "Custom role: $RoleName for alert reading created successfully with role id: $RoleId."
+      Remove-Item -Path $CustomRoleJsonFilePath
+  }
+  else {
+      Log "Custom role: $RoleName for alert reading already exists with role id: $RoleId."
+  }
+
+  Log "Assigning the custom role: $RoleName to the reporting app: $ConnectorAppName"
+  $ConnectorPrincipalId = (Get-AzADServicePrincipal -DisplayName $ConnectorAppName).id
+
+  $AssignedCustomRole = Get-AzRoleAssignment -ObjectId $ConnectorPrincipalId -RoleDefinitionId $RoleId -Scope "/subscriptions/$SubscriptionId"
+
+  if (-not $AssignedCustomRole) {
+      New-AzRoleAssignment -ObjectId $ConnectorPrincipalId -RoleDefinitionId $RoleId -Scope "/subscriptions/$SubscriptionId"
+      Log "Custom role: $RoleName assigned to the reporting app: $ConnectorAppName"
+  }
+  else {
+      Log "Custom role: $RoleName already assigned to the reporting app: $ConnectorAppName"
+  }
+
+  Log "Successfully setup permissions for alert reading"
+}
+
+function Deploy-Connector {
```

Comment:
Nit: Can we call this Create-Container, and change it in the logs everywhere too?
Deploying gives a very code deployment feel, and for the customers, they're just creating a resource. Like creating a VM. And it would also be consistent with the portal terminology.

Reply:
Discussed offline that both terminologies are visible on the portal for a resource. 
@bhavanakonchada what are your thoughts on this?

Reply:
Done